lnrpc/routerrpc/router_backend.go

```diff
@@ -547,6 +547,14 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 	}
 	payIntent.CltvLimit = cltvLimit
 
+	// Take max htlcs from the request. Map zero to one for backwards
+	// compatibility.
+	maxHtlcs := rpcPayReq.MaxHtlcs
+	if maxHtlcs == 0 {
+		maxHtlcs = 1
+	}
+	payIntent.MaxHtlcs = maxHtlcs
```

Comment:
The check in this method to only allow values below the max payment size should be removed.

Reply:
Can that be done in an independent pr? It isn't necessary for mpp and also has value for single-shot payments.